### ISSUE TITLE
Recurring events are not always synched correctly to WP

### ISSUE THREAD

**Issue by a-schild** [a-schild/churchtools-wp-calendarsync#2]
Needs to process the additions or the repeatId, and probably use the additions->id's too in the sync table

```
   {
      "base": {
        "id": 600783,
        "caption": "Nidau: Jungschar-Nachmittag",
        "note": "EL",
        "address": {
          "meetingAt": "Bibliotheksplatz Nidau",
          "street": "Hauptstrasse 73",
          "addition": null,
          "district": null,
          "zip": "2560",
          "city": "Nidau",
          "country": "CH",
          "latitude": "47.1244565",
          "longitude": "7.2411075"
        },
        "version": 1,
        "calendar": {
          "id": 32,
          "name": "Anlässe FW und extern",
          "nameTranslated": "Anlässe FW und extern",
          "sortKey": 14,
          "color": "#f691b2",
          "isPublic": true,
          "isPrivate": false,
          "randomUrl": "Y4qjxtxPOAFutQ4A7cLriupJlaFMYvuw",
          "iCalSourceUrl": "",
          "campusId": null,
          "eventTemplateId": null,
          "meta": {
            "modifiedDate": "2018-01-24T14:22:11Z",
            "modifiedPid": 1
          }
        },
        "information": "Treffpunkt beim Bibliotheksplatz Nidau\nKontakt:\njsnidau@cevi.ch\n",
        "image": null,
        "link": "",
        "isInternal": false,
        "startDate": "2023-11-04T13:00:00Z",
        "endDate": "2023-11-04T16:15:00Z",
        "allDay": false,
        "repeatId": 999,
        "repeatFrequency": null,
        "repeatUntil": null,
        "repeatOption": null,
        "additions": [
          {
            "id": 1107,
            "date": "2023-11-18",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          },
          {
            "id": 1108,
            "date": "2023-12-02",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          },
          {
            "id": 1109,
            "date": "2023-12-16",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          }
        ],
        "exceptions": [],
        "signup": null,
        "meta": {
          "createdDate": "2023-09-05T15:50:09Z",
          "createdPerson": {
            "id": 1516
          },
          "modifiedDate": "2023-09-28T07:28:50Z",
          "modifiedPerson": {
            "id": 1
          }
        },
        "onBehalfOfPid": null
      },
      "calculated": {
        "startDate": "2023-11-04T13:00:00Z",
        "endDate": "2023-11-04T16:15:00Z"
      }
    }
```

**Comment by a-schild** [comment]
Needs to process the additions or the repeatId, and probably use the additions->id's too in the sync table
```
   {
      "base": {
        "id": 600783,
        "caption": "Nidau: Jungschar-Nachmittag",
        "note": "EL",
        "address": {
          "meetingAt": "Bibliotheksplatz Nidau",
          "street": "Hauptstrasse 73",
          "addition": null,
          "district": null,
          "zip": "2560",
          "city": "Nidau",
          "country": "CH",
          "latitude": "47.1244565",
          "longitude": "7.2411075"
        },
        "version": 1,
        "calendar": {
          "id": 32,
          "name": "Anlässe FW und extern",
          "nameTranslated": "Anlässe FW und extern",
          "sortKey": 14,
          "color": "#f691b2",
          "isPublic": true,
          "isPrivate": false,
          "randomUrl": "Y4qjxtxPOAFutQ4A7cLriupJlaFMYvuw",
          "iCalSourceUrl": "",
          "campusId": null,
          "eventTemplateId": null,
          "meta": {
            "modifiedDate": "2018-01-24T14:22:11Z",
            "modifiedPid": 1
          }
        },
        "information": "Treffpunkt beim Bibliotheksplatz Nidau\nKontakt:\njsnidau@cevi.ch\n",
        "image": null,
        "link": "",
        "isInternal": false,
        "startDate": "2023-11-04T13:00:00Z",
        "endDate": "2023-11-04T16:15:00Z",
        "allDay": false,
        "repeatId": 999,
        "repeatFrequency": null,
        "repeatUntil": null,
        "repeatOption": null,
        "additions": [
          {
            "id": 1107,
            "date": "2023-11-18",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          },
          {
            "id": 1108,
            "date": "2023-12-02",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          },
          {
            "id": 1109,
            "date": "2023-12-16",
            "isRepeated": false,
            "meta": {
              "modifiedDate": "2023-09-05T15:50:09Z",
              "modifiedPid": 1516
            }
          }
        ],
        "exceptions": [],
        "signup": null,
        "meta": {
          "createdDate": "2023-09-05T15:50:09Z",
          "createdPerson": {
            "id": 1516
          },
          "modifiedDate": "2023-09-28T07:28:50Z",
          "modifiedPerson": {
            "id": 1
          }
        },
        "onBehalfOfPid": null
      },
      "calculated": {
        "startDate": "2023-11-04T13:00:00Z",
        "endDate": "2023-11-04T16:15:00Z"
      }
    }
```

**Comment by a-schild** [comment]
Nope, use these fields for start/end timestamps and not the regular start/end dates from the main entry/body

"calculated": {
        "startDate": "2023-11-04T13:00:00Z",
        "endDate": "2023-11-04T16:15:00Z"
      }

**Comment by a-schild** [comment]
https://github.com/5pm-HDH/churchtools-api/issues/176

**Comment by a-schild** [comment]
Fixed in v1.0.1
